binrpc/binrpc.go

```diff
@@ -96,22 +96,22 @@ func ConstructHeader(header *bytes.Buffer, payloadLength uint64, cookie uint8) e
 	log.Printf("Payload length is %d, and the length of that value in bytes is %d", payloadLength, plSize)
 
 	// Find the size of the cookie
-	cookieSize := binary.Size(uint8(cookie))
+	cookieSize := binary.Size(cookie)
 	if cookieSize < 0 {
 		return fmt.Errorf("Failed to determine byte length of cookie")
 	}
 
 	// Write the Flags/LL/CL byte (flags hard-coded to 0x0 for now)
-	err = header.WriteByte(byte(0x0<<4 | uint(plSize)<<2 | uint(cookieSize)))
+	err = header.WriteByte(byte(0x0<<4 | uint(plSize-1)<<2 | uint(cookieSize-1)))
 
 	// Write the payload length
-	err = binary.Write(header, binary.LittleEndian, payloadLength)
+	err = binary.Write(header, binary.BigEndian, uint8(payloadLength))
```

Comment:
x86 is little endian... does the binrpc protocol actually actively encode using big endian?